ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -371,12 +376,68 @@ def add_capacity(osd_size_capacity_requested):
     storageDeviceSets->count = (capacity reqested / osd capacity ) + existing count storageDeviceSets
 
     """
-
+    osd_size_existing = get_osd_size()
+    device_sets_required = int(osd_size_capacity_requested / osd_size_existing)
+    lvpresent = check_local_volume()
+    if lvpresent:
+        ocp_obj = OCP()
+        workers = get_typed_nodes(node_type='worker')
+        worker_names = [worker.name for worker in workers]
+        output = ocp_obj.exec_oc_cmd("get localvolume local-block -n local-storage -o yaml")
+        cur_device_list = output["spec"]["storageClassDevices"][0]['devicePaths']
+        path = os.path.join(constants.EXTERNAL_DIR, "device-by-id-ocp")
+        utils.clone_repo(constants.OCP_QE_DEVICEPATH_REPO, path)
+        os.chdir(path)
+        utils.run_cmd("ansible-playbook devices_by_id.yml")
+        f = open("local-storage-block.yaml", "r")
+        f2 = open("local-block.yaml", "w")
+        w1, w2, w3 = device_sets_required, device_sets_required, device_sets_required
+        cur_line = f.readline()
+        while "devicePaths:" not in cur_line:
+            f2.write(cur_line)
+            cur_line = f.readline()
+        f2.write(cur_line)
+        cur_line = f.readline()
+        while cur_line:
+            if str(osd_size_capacity_requested) in cur_line:
+                if w1 and (str(worker_names[0]) in cur_line):
+                    if not any(s in cur_line for s in cur_device_list):
+                        f2.write(cur_line)
+                        w1 = w1 - 1
+                if w2 and (str(worker_names[1]) in cur_line):
+                    if not any(s in cur_line for s in cur_device_list):
+                        f2.write(cur_line)
+                        w2 = w2 - 1
+                if w3 and (str(worker_names[2]) in cur_line):
+                    if not any(s in cur_line for s in cur_device_list):
+                        f2.write(cur_line)
+                        w3 = w3 - 1
+            cur_line = f.readline()
+        f.close()
+        f2.close()
+        obj = open("local-block.yaml")
```

Comment:
Again, `obj` isn't very descriptive. Let's try to come up with something more representative. `local_block_yaml`?

Reply:
Added more descriptive names!